### PR TITLE
Fix backslash unescaping logic

### LINKED ISSUE #270: `unescape` removes escaped backslashes
`unescape(…)` | Expected | Actual
--- | --- | ---
`\` | `\` | `\` ✓
`\\` | `\` | `\` ✓
`\\\\` | `\\` | `\` ✗
`\\\*` | `\*` | `*` ✗

### PR DESCRIPTION
**Pull request by raphlinus**
Fixes #270

### CODE REVIEW

**Comment by raphlinus** [comment]
This simplifies some existing code, so it makes sense to check that I've got it right. In particular, the title in a refdef scan doesn't get changed, so it can be borrowed from the source string rather than being stored as a cow.

File: src/parse.rs [hunk 1]
```diff
@@ -1319,31 +1311,12 @@ impl<'a> FirstPass<'a> {
                 }
                 '\\' => {
```
Comment (marcusklaas) [inline src/parse.rs]: I think we can skip this branch altogether now, no?

**Comment by marcusklaas** [comment]
Nice cleanup! Just one request: since this wasn't caught by the spec, can we add a regression test for this ourselves?

File: src/parse.rs (hunk 1 above)
Comment (raphlinus) [reply src/parse.rs]: Pretty sure not, you have to resolve the difference between `(title\))` and `(title))`. The former is a valid link title, the second isn't. As you say, tests will help settle the question, I'll add those.

**Pull request merged**